library/core/src/primitive_docs.rs

```diff
@@ -1273,10 +1276,10 @@ mod prim_f16 {}
 /// (e.g. `min`, `minimum`, `max`, `maximum`); other aspects of their semantics and which IEEE 754
 /// operation they correspond to are documented with the respective functions.
 ///
-/// When a floating-point operation is executed in `const` context, the same rules apply: no
-/// guarantee is made about which of the NaN bit patterns described above will be returned. The
-/// result does not have to match what happens when executing the same code at runtime, and the
-/// result can vary depending on factors such as compiler version and flags.
+/// When an arithmetic floating point operation is executed in `const` context, the same rules
```

Comment:
The hyphen is actually correct. Consider encountering:
- a friendly-looking logician, versus
- a friendly little logician

Notice that "looking logician" doesn't hold up as a descriptor, whereas "little logician" does, and that's why they get hyphenated.
```suggestion
/// When an arithmetic floating-point operation is executed in `const` context, the same rules
```

Also consider the titles here:
- https://en.wikipedia.org/wiki/Floating-point_arithmetic
- https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html

Reply:
But also this is actually probably supposed to be
```suggestion
/// When a floating-point arithmetic operation is executed in `const` context, the same rules
```

Reply:
I removed the hyphen since the rest of the file doesn't use a hyphen. I can also add it everywhere.

For the 2nd one, the order of adjectives in English follows arcane rules so I won't argue about it. ;)

Reply:
Hm actually no I think I want to argue. The emphasis here is on "arithmetic", so putting that first seems more clear to me. This distinction of "arithmetic" and "non-arithmetic" is also a floating-point thing, so it's not like we have an "arithmetic operation" in its floating-point variant.

So, "arithmetic floating-point operation" makes more sense to me.

Reply:
Hm, okay!